Framework/Algorithms/test/MaxEntTest.h

```diff
@@ -110,6 +110,80 @@ class MaxEntTest : public CxxTest::TestSuite {
+    int nHist = 6;
+    int nBins = 10;
+    auto ws = WorkspaceCreationHelper::create2DWorkspace(nHist, nBins);
+
+    IAlgorithm_sptr alg = AlgorithmManager::Instance().create("MaxEnt");
+    alg->initialize();
+    alg->setChild(true);
+    alg->setProperty("InputWorkspace", ws);
+    alg->setProperty("ComplexData", true);
+    alg->setPropertyValue("MaxIterations", "1");
+    alg->setProperty("DataLinearAdj", ws);
+    alg->setProperty("DataConstAdj", ws);
+    alg->setPropertyValue("ReconstructedImage", "image");
+    alg->setPropertyValue("ReconstructedData", "data");
+    alg->setPropertyValue("EvolChi", "evolChi");
+    alg->setPropertyValue("EvolAngle", "evolAngle");
+
+    TS_ASSERT_THROWS_NOTHING(alg->execute());
+
+    MatrixWorkspace_sptr data = alg->getProperty("ReconstructedData");
+    MatrixWorkspace_sptr image = alg->getProperty("ReconstructedImage");
+    MatrixWorkspace_sptr chi = alg->getProperty("EvolChi");
+    MatrixWorkspace_sptr angle = alg->getProperty("EvolAngle");
+
+    TS_ASSERT_EQUALS(data->getNumberHistograms(), nHist);
+    TS_ASSERT_EQUALS(image->getNumberHistograms(), nHist);
+    TS_ASSERT_EQUALS(chi->getNumberHistograms(), nHist / 2);
+    TS_ASSERT_EQUALS(angle->getNumberHistograms(), nHist / 2);
+
+    TS_ASSERT_EQUALS(data->blocksize(), nBins);
+    TS_ASSERT_EQUALS(image->blocksize(), nBins);
+    TS_ASSERT_EQUALS(chi->blocksize(), 1);
+    TS_ASSERT_EQUALS(angle->blocksize(), 1);
+  }
+
+  void test_complex_data_adjustments_together() {
+    int nHist = 6;
+    int nBins = 10;
+    auto ws = WorkspaceCreationHelper::create2DWorkspace(nHist, nBins);
+
+    IAlgorithm_sptr alg = AlgorithmManager::Instance().create("MaxEnt");
+    alg->initialize();
+    alg->setChild(true);
+    alg->setProperty("InputWorkspace", ws);
+    alg->setProperty("ComplexData", true);
+    alg->setPropertyValue("MaxIterations", "1");
+    alg->setProperty("DataLinearAdj", ws);
+    alg->setProperty("DataConstAdj", ws);
+    alg->setProperty("PerSpectrumReconstruction", false);
+    alg->setPropertyValue("ReconstructedImage", "image");
+    alg->setPropertyValue("ReconstructedData", "data");
+    alg->setPropertyValue("EvolChi", "evolChi");
+    alg->setPropertyValue("EvolAngle", "evolAngle");
+
+    TS_ASSERT_THROWS_NOTHING(alg->execute());
+
+    MatrixWorkspace_sptr data = alg->getProperty("ReconstructedData");
+    MatrixWorkspace_sptr image = alg->getProperty("ReconstructedImage");
+    MatrixWorkspace_sptr chi = alg->getProperty("EvolChi");
+    MatrixWorkspace_sptr angle = alg->getProperty("EvolAngle");
+
+    TS_ASSERT_EQUALS(data->getNumberHistograms(), nHist);
+    TS_ASSERT_EQUALS(image->getNumberHistograms(), 2);
+    TS_ASSERT_EQUALS(chi->getNumberHistograms(), 1);
+    TS_ASSERT_EQUALS(angle->getNumberHistograms(), 1);
+
+    TS_ASSERT_EQUALS(data->blocksize(), nBins);
+    TS_ASSERT_EQUALS(image->blocksize(), nBins);
+    TS_ASSERT_EQUALS(chi->blocksize(), 1);
+    TS_ASSERT_EQUALS(angle->blocksize(), 1);
+  }
```

Comment:
we should also check the values when using the new adjustments

Reply:
we should also check the simple case

Reply:
Why?

Reply:
if this one fails we know the bug is in adjustment. If ```test_adjustments_three_spectra_together``` fails and the simple case does not then we know the bug is in the combining of the spectra. At present we would not know if its the combining or the adjustment.

Reply:
I don't see any sense in testing values in tests where the maximumum number of iterations has been set to 1.

Reply:
could we have a test (or system test) with more iterations? Does ```test_adjustments_three_spectra_together ``` just have 1 iteration too?

Reply:
Doesn't test `test_three_spectra_together` take care of the case where the spectra are just combined and there are no adjustments?

Reply:
I've got tests for adjustments too.

Reply:
Do you want a test that explicitly tests the arithmetic of the adjustments rather than showing that the fitting works with adjustments and gives certain data values not explicitly determined by adjustment arithmetic?

Reply:
I note that MaxentTransformMultiFourier does explicitly test the adjustment arithmestic (but with real and complex adjustmests separately to keep it simple). So any such test in Maxent would merely confirm the the adjusted values were delivered correctly to the data workspace.

Reply:
that is still useful information.

Reply:
So I I'll go ahead with a cursory test of Maxent the explicitly tests the arithmetic.

Reply:
The test took most of the day to write, but has shown a problem with to `toComplex` function in `MaxEnt`. It interleaves the spectra instead of concatenating them as required. I'll be away until Wednesday next week, then I'll fix it.

Reply:
Work on this has been completed and the issue is ready for testing again.